CharacterGenerator/CharacterGenerator/CreateSoldierTableViewController.m

```diff
@@ -0,0 +1,317 @@
+//
+//  CreateSoldierTableViewController.m
+//  CharacterGenerator
+//
+//  Created by Z on 8/2/15.
+//  Copyright (c) 2015 Mike Kavouras. All rights reserved.
+//
+
+#import "CreateSoldierTableViewController.h"
+#import "COD4database.h"
+
+@interface CreateSoldierTableViewController ()
+
+@property (weak, nonatomic) IBOutlet UITextField *nameTextField;
+
+@property (nonatomic) COD4database *model;
+
+@end
+
+@implementation CreateSoldierTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    //setup model
+    self.model = [[COD4database alloc] init];
+    [self.model setup];
+    
+    // Connect data
+    self.teamPicker.dataSource = self;
+    self.teamPicker.delegate = self;
+    
+    self.primaryWeaponTypePicker.dataSource = self;
+    self.primaryWeaponTypePicker.delegate = self;
+    
+    self.primaryWeaponPicker.dataSource = self;
+    self.primaryWeaponPicker.delegate = self;
+    
+    self.secondaryWeaponTypePicker.dataSource = self;
+    self.secondaryWeaponTypePicker.delegate = self;
+    
+    self.secondaryWeaponPicker.dataSource = self;
+    self.secondaryWeaponPicker.delegate = self;
+    
+    self.perk1Picker.dataSource = self;
+    self.perk1Picker.delegate = self;
+    
+    self.perk2Picker.dataSource = self;
+    self.perk2Picker.delegate = self;
+    
+    self.perk3Picker.dataSource = self;
+    self.perk3Picker.delegate = self;
+    
+}
+
+-(void)saveRecruit{
+    
+    Soldier *recruit = [[Soldier alloc] init];
+    recruit.name = self.nameTextField.text;
+    
+    NSInteger teamRow = [self.teamPicker selectedRowInComponent:0];
+    recruit.team = [[self.model.team allKeys] objectAtIndex:teamRow];
+    
+    NSInteger primaryWeaponTypeRow = [self.primaryWeaponTypePicker selectedRowInComponent:0];
+    recruit.primaryWeaponType = [[self.model.primaryWeapons allKeys] objectAtIndex:primaryWeaponTypeRow];
+    
+    NSInteger primaryWeaponRow = [self.primaryWeaponPicker selectedRowInComponent:0];
+    NSArray *primaryWeaponsOfType = [self.model.primaryWeapons objectForKey:recruit.primaryWeaponType];
+    recruit.primaryWeapon = [primaryWeaponsOfType objectAtIndex:primaryWeaponRow];
+    
+    NSInteger secondaryWeaponTypeRow = [self.secondaryWeaponTypePicker selectedRowInComponent:0];
+    recruit.secondaryWeaponType = [[self.model.secondaryWeapons allKeys] objectAtIndex:secondaryWeaponTypeRow];
+    
+    NSInteger secondaryWeaponRow = [self.secondaryWeaponPicker selectedRowInComponent:0];
+    NSArray *secondaryWeaponsOfType = [self.model.secondaryWeapons objectForKey:recruit.secondaryWeaponType];
+    recruit.secondaryWeapon = [secondaryWeaponsOfType objectAtIndex:secondaryWeaponRow];
+    
+    recruit.specialGrenade = [self.model.specialGrenades objectAtIndex:self.specialGrenadeSegmentedControl.selectedSegmentIndex];
```

Comment:
If the user doesn't choose a special grenade and then goes to add a soldier to the roster the app will crash. Reason being that `self.specialGrenadeSegmentedControl.selectedSegmentIndex` does not return a valid value until a segment is actually selected so you're try to access the object in the `specialGrenades array at a invalid index. 

The crash looks like this in the debugger...it's probably a familiar one :)
`2015-08-10 20:41:34.728 CharacterGenerator[20484:9965180] *** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSArrayI objectAtIndex:]: index 18446744073709551615 beyond bounds [0 .. 2]'`

The solution here could be to force the user to choose a grenade before allowing them to tap the `Add to Roster` by disabling the button or to choose a default grenade for them by setting the value of `self.specialGrenadeSegmentedControl.selectedSegmentIndex` during set up.